File: src/main/java/io/vproxy/vfx/manager/audio/AudioWrapper.java
package io.vproxy.vfx.manager.audio;

import io.vproxy.vfx.util.FXUtils;
import javafx.scene.media.AudioClip;

public class AudioWrapper {
    private final AudioClip clip;
    private int count = 0;
    private boolean lastPlayed = false;

    public AudioWrapper(AudioClip clip) {
        this.clip = clip;
    }

    public int getCount() {
        return count;
    }

    public void play() {
        FXUtils.runOnFX(() -> {
            ++count;
            clip.play();
        });
    }

    public boolean isLastPlayed() {
        return lastPlayed;
    }

    public void setLastPlayed(boolean lastPlayed) {
        this.lastPlayed = lastPlayed;
    }
}

File: src-ts/src/manager/audio/AudioWrapper.ts
import type { AudioClip } from './AudioManager.js';
import { FXUtils } from '../../util/FXUtils.js';

export class AudioWrapper {
  private readonly clip: AudioClip;
  private count = 0;
  private lastPlayed = false;

  constructor(clip: AudioClip) {
    this.clip = clip;
  }

  getCount(): number { return this.count; }

  play(): void {
    FXUtils.runOnFX(() => {
      this.count += 1;
      // Reset to start so repeated plays restart the clip.
      try {
        this.clip.currentTime = 0;
        void this.clip.play();
      } catch {
        // autoplay policy may reject
      }
    });
  }

  isLastPlayed(): boolean { return this.lastPlayed; }
  setLastPlayed(v: boolean): void { this.lastPlayed = v; }
}
